base_layer/core/src/blocks/block_header.rs

```diff
@@ -168,16 +169,34 @@ impl BlockHeader {
         }
     }
 
+    /// Returns the [std::cmp::Ordering] of the timestamp of this header compared to the "other" header
+    pub fn compare_timestamp(&self, other: &BlockHeader) -> Ordering {
+        self.timestamp.as_u64().cmp(&other.timestamp.as_u64())
+    }
+
     #[cfg(feature = "base_node")]
     pub fn into_builder(self) -> BlockBuilder {
         BlockBuilder::new(self.version).with_header(self)
     }
 
-    /// Given a slice of headers (in reverse order), calculate the maximum, minimum and average periods between them
+    /// Given a slice of headers, calculate the maximum, minimum and average periods between them.
+    /// Allocates in case the headers are in the wrong order.
     pub fn timing_stats(headers: &[BlockHeader]) -> (u64, u64, f64) {
         if headers.len() < 2 {
             (0, 0, 0.0)
         } else {
+            let mut headers = headers.to_vec();
+
+            // ensure the slice is in reverse order
+            let ordering = headers[0].compare_timestamp(&headers[headers.len() - 1]);
```

Comment:
nit: I don't think the compare_timestamp is generally useful enough
```suggestion
            let ordering = headers[0].timestamp.cmp(&headers[headers.len() - 1].timestamp);
```